### DISCUSSION TITLE
Register listener for job created events

### DISCUSSION THREAD

**Discussion by jmxti** [jobrunr/jobrunr#339] [Ideas]
You can attach listeners to the `StorageProvider` for a variety of events. But I cannot find any way to listen for the creation of new jobs. Something like:

```java
storageProvider.addJobStorageOnChangeListener (new JobCreateListener () {
    @Override public void onCreate (Job job) {
        System.out.println ("Created a new job with id: " + job.getId ());
    }
});
```

where `JobCreateListener` might be something like:

```java
public interface JobCreateListener extends StorageProviderChangeListener {
     public void onCreate (Job job);
}
```

Is there a reason why this is not available? Is there some technical limitation that prevents such a thing?

ps: I would be willing to contribute code if the 'feature' is possible and desirable.

**Comment by rdehuyss** [comment]
What would be the exact use-case? As you are already creating the job and submitting it to JobRunr, I do not completely understand the added value of this filter. Probably because I don't know your use-case :-)
